tasks/install-package.yml

```diff
@@ -0,0 +1,114 @@
+- name: install the rpm repo's key
+  rpm_key:
+    state: present
+    key: "{{ rpm_repo_key }}"
+  when: ansible_os_family == "RedHat"
+
+- name: install apt repo keys
+  apt_key:
+    state: present
+    url: "{{ item.url }}"
+    id: "{{ item.id }}"
+  when: ansible_os_family == "Debian"
+  loop:
+  - { url: "{{ deb_repo_key }}", id: "{{ deb_repo_key_id }}" }
+  - { url: "https://deb.nodesource.com/gpgkey/nodesource.gpg.key", id: "1655A0AB68576280" }
+
+- name: install the rpm repo
+  package:
+    name: "{{ rpm_repo_url }}"
+    state: present
+    disable_gpg_check: "{{ disable_gpg_check_rpm_repo | bool }}"
+  when: ansible_os_family == "RedHat"
+
+- name: install the apt repo
+  apt:
+    deb: "{{ apt_repo_url }}"
+    state: present
+    update_cache: "{{ apt_update_cache | bool }}"
+  when: ansible_os_family == "Debian"
+
+- name: remove node 14 repo if it exists
+  file:
+    path: /etc/apt/sources.list.d/nodesource.list
+    state: absent
+  when: ansible_os_family == "Debian"
+
+- name: install node's apt repo
+  apt_repository:
+    repo: "{{ item }}"
+    state: present
+    filename: nodesource
+    update_cache: true
+  loop:
+    - "deb https://deb.nodesource.com/node_12.x {{ ansible_distribution_release }} main"
+    - "deb-src https://deb.nodesource.com/node_12.x {{ ansible_distribution_release }} main"
+  when: ansible_os_family == "Debian"
+
+- name: DELME HACK FOR DEBIAN TESTING
+  shell: sed -i 's#latest#build/2.0#g' /etc/apt/sources.list.d/ondemand-web.list
+  when: ansible_os_family == "Debian"
+
+- name: enable epel
+  package:
+    name: 'epel-release'
+    state: present
+  when: ansible_os_family == "RedHat"
+
+- name: install rhel/centos:8 dependencies
```

Comment:
You also need 

```
dnf module enable ruby:3.0
dnf module enable nodejs:14
```

If those modules are already enabled you must do extra steps:

```
dnf module reset nodejs
dnf module enable nodejs:14
dnf module reset ruby
dnf module enable ruby:3.0
```

Reply:
The DNF modules only needed on EL8.  Seems EL9 started with no modules so Ruby 3.0 and NodeJS 14 or 16 are default without modules.

Reply:
They pick these up from files in `vars/`, but you're right the way we have Ubuntu files, we likely need more var files for 7, 8 & 9.

https://github.com/OSC/ood-ansible/blob/9dc82d7b4cae1ea40e6e83d81722d304c7164023/vars/RedHat.yml#L14-L17

Reply:
Ah, shows my lack of Ansible knowledge.  Ya I think files for 7, 8 and 9 (or 7 and 8 till support 2.1) makes sense.